src/server.c

```diff
@@ -2417,7 +2461,7 @@ dictType dbExpiresDictType = {
 
 /* Command table. sds string -> command struct pointer. */
 dictType commandTableDictType = {
-    dictSdsCaseHash,            /* hash function */
+    dictPresetSeedSdsCaseHash,  /* hash function */
```

Comment:
how did you conclude which dictTypes should be the ones that get the preset seed and which ones don't?
i'm afraid you may miss some (or that new ones will be added in the future, and we won't notice the problem).
maybe we need some automatic way / or assertion?

also, there could be a case where there's a dictType that serves two dicts, one set at start time and one doesn't.
in that case we can get either one of these:
1. they both use the non-preset seed, and then maybe the test or assert can catch it.
2. they both use the preset seed, and we won't notice (or care?)

maybe we should add a test that verifies that at least the main dict, and also the hash-type dict have random seed?

Reply:
Verified by code review and a temp assert in the hash function. I considered leaving something more permanent there, but decided against that:
* Avoid more clutter
* Likelyhood of this happening is pretty low, and happening and not being caught is even smaller.

As you can see there there are really very few dictTypes that are used so early on startup (basically just command table, module API and modules) and they are not the generic types. Even if they did get reused - the only problem it would create is that the `hash-seed` will not apply for them so they'll always be non-deterministic.

Reply:
what about a case where a dict gets partially populated with one seed, and then the seed changes after it has data?

and what about the case where one of the dicts that is being populated at runtime (like one for set type) will accidentally use the preset dict? Raft will have inconsistencies in scripts, and you're unlikely to catch that bug.

Reply:
As a rule preset should not be used, except for very specific cases. I agree it's not perfect, but I suppose it's better than the alternative of keeping the seed as part of every dict instance (dictType is not good enough of course).

Reply:
so could the two cases above cases happen if we mis-label the dict types?
obviously case 2 can happen, but since you're saying we almost never use this type, it's not a real concern.

and case 1 will only happen if for some reason we call `initDictHashSeed` for a second time with `NULL` as input, right?
so that's not the case, but maybe we should made a slight change in the code to add the protection against that in `initDictHashSeed` itself and not rely on the caller?